<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Effector Analysis with paneffectR • paneffectR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Effector Analysis with paneffectR">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">paneffectR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes">
<li><a class="dropdown-item" href="../articles/getting-started.html">Getting Started</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Workflows</h6></li>
    <li><a class="dropdown-item" href="../articles/effector-analysis.html">Effector Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/pan-genome.html">Pan-Genome Analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Technical</h6></li>
    <li><a class="dropdown-item" href="../articles/algorithms.html">Algorithm Deep Dive</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/TeamMacLean/paneffectR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Effector Analysis with paneffectR</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/TeamMacLean/paneffectR/blob/main/vignettes/effector-analysis.Rmd" class="external-link"><code>vignettes/effector-analysis.Rmd</code></a></small>
      <div class="d-none name"><code>effector-analysis.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette demonstrates how to use paneffectR with output from the
<a href="https://github.com/TeamMacLean/omnieff" class="external-link">omnieff</a> effector
prediction pipeline. A key focus is <strong>identifying singleton
effectors</strong> - proteins unique to individual isolates that may
drive host specificity or represent recently evolved virulence
factors.</p>
</div>
<div class="section level2">
<h2 id="setup-and-data-loading">Setup and Data Loading<a class="anchor" aria-label="anchor" href="#setup-and-data-loading"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/TeamMacLean/paneffectR" class="external-link">paneffectR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></code></pre></div>
<p>For this vignette, we use pre-computed clustering results with
synthetic data that simulates typical effector analysis:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">visual_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"testdata"</span>, <span class="st">"visual"</span>, package <span class="op">=</span> <span class="st">"paneffectR"</span><span class="op">)</span></span>
<span><span class="va">clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">visual_dir</span>, <span class="st">"clusters_visual.rds"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">proteins</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">visual_dir</span>, <span class="st">"proteins_visual.rds"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">clusters</span></span>
<span><span class="co">#&gt; -- orthogroup_result (synthetic_visual) --</span></span>
<span><span class="co">#&gt; 50 orthogroups</span></span>
<span><span class="co">#&gt; 12 singletons</span></span>
<span><span class="va">proteins</span></span>
<span><span class="co">#&gt; -- protein_collection --</span></span>
<span><span class="co">#&gt; 6 assemblies, 172 total proteins</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 3</span></span></span>
<span><span class="co">#&gt;   assembly_name n_proteins has_scores</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> asm_A                 25 TRUE      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> asm_B                 31 TRUE      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> asm_C                 27 TRUE      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> asm_D                 31 TRUE      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> asm_E                 31 TRUE      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> asm_F                 27 TRUE</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="finding-unique-effectors-singletons">Finding Unique Effectors: Singletons<a class="anchor" aria-label="anchor" href="#finding-unique-effectors-singletons"></a>
</h2>
<p>When comparing effector repertoires across pathogen isolates,
<strong>singletons are often the most biologically interesting
proteins</strong>. These are proteins that couldn’t be clustered with
any protein from other assemblies - they appear to be unique to a single
isolate.</p>
<p>Singletons are important because they may represent:</p>
<ul>
<li>
<strong>Recently evolved effectors</strong> - Novel proteins that
emerged in a specific lineage</li>
<li>
<strong>Candidate avirulence genes</strong> - Effectors recognized
by host resistance proteins, driving isolate-specific
incompatibility</li>
<li>
<strong>Host adaptation factors</strong> - Proteins that enable
colonization of specific host genotypes</li>
<li>
<strong>Candidates for functional characterization</strong> -
High-priority targets for understanding what makes isolates
different</li>
</ul>
<div class="section level3">
<h3 id="extracting-singletons">Extracting Singletons<a class="anchor" aria-label="anchor" href="#extracting-singletons"></a>
</h3>
<p>paneffectR provides dedicated functions for working with
singletons:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get all singletons</span></span>
<span><span class="va">singletons</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_singletons.html">get_singletons</a></span><span class="op">(</span><span class="va">clusters</span><span class="op">)</span></span>
<span><span class="va">singletons</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 12 × 2</span></span></span>
<span><span class="co">#&gt;    protein_id         assembly</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> asm_A_singleton_01 asm_A   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> asm_B_singleton_02 asm_B   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> asm_C_singleton_03 asm_C   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> asm_D_singleton_04 asm_D   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> asm_E_singleton_05 asm_E   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> asm_F_singleton_06 asm_F   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> asm_A_singleton_07 asm_A   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> asm_B_singleton_08 asm_B   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> asm_C_singleton_09 asm_C   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> asm_D_singleton_10 asm_D   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> asm_E_singleton_11 asm_E   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> asm_F_singleton_12 asm_F</span></span>
<span></span>
<span><span class="co"># Count singletons</span></span>
<span><span class="fu"><a href="../reference/n_singletons.html">n_singletons</a></span><span class="op">(</span><span class="va">clusters</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 12</span></span>
<span></span>
<span><span class="co"># Summary by assembly</span></span>
<span><span class="fu"><a href="../reference/singletons_by_assembly.html">singletons_by_assembly</a></span><span class="op">(</span><span class="va">clusters</span>, <span class="va">proteins</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 4</span></span></span>
<span><span class="co">#&gt;   assembly n_singletons n_total pct_singleton</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> asm_A               2      25          8   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> asm_B               2      31          6.45</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> asm_C               2      27          7.41</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> asm_D               2      31          6.45</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> asm_E               2      31          6.45</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> asm_F               2      27          7.41</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="singleton-scores-are-they-real-effectors">Singleton Scores: Are They Real Effectors?<a class="anchor" aria-label="anchor" href="#singleton-scores-are-they-real-effectors"></a>
</h3>
<p>A key question is whether singletons are genuine effector candidates
or just annotation artifacts. We can examine their prediction scores to
assess confidence:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get all proteins with scores</span></span>
<span><span class="va">all_proteins</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">proteins</span><span class="op">$</span><span class="va">assemblies</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">asm</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">proteins</span><span class="op">$</span><span class="va">assemblies</span><span class="op">[[</span><span class="va">asm</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">proteins</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>assembly <span class="op">=</span> <span class="va">asm</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Join singletons with their scores</span></span>
<span><span class="va">singleton_scores</span> <span class="op">&lt;-</span> <span class="va">singletons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span></span>
<span>    <span class="va">all_proteins</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">protein_id</span>, <span class="va">custom_score</span><span class="op">)</span>,</span>
<span>    by <span class="op">=</span> <span class="st">"protein_id"</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">custom_score</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">singleton_scores</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 12 × 3</span></span></span>
<span><span class="co">#&gt;    protein_id         assembly custom_score</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> asm_C_singleton_03 asm_C            8.89</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> asm_B_singleton_02 asm_B            8.76</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> asm_A_singleton_01 asm_A            7.18</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> asm_D_singleton_04 asm_D            6.70</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> asm_F_singleton_06 asm_F            5.71</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> asm_E_singleton_05 asm_E            5.17</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> asm_B_singleton_08 asm_B            4.89</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> asm_A_singleton_07 asm_A            4.81</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> asm_C_singleton_09 asm_C            2.87</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> asm_D_singleton_10 asm_D            2.70</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> asm_F_singleton_12 asm_F            2.64</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> asm_E_singleton_11 asm_E            2.16</span></span></code></pre></div>
<p>Several singletons have high effector prediction scores (&gt;6),
suggesting they are genuine effector candidates worthy of follow-up, not
just noise in the data.</p>
</div>
<div class="section level3">
<h3 id="visualizing-singleton-score-distribution">Visualizing Singleton Score Distribution<a class="anchor" aria-label="anchor" href="#visualizing-singleton-score-distribution"></a>
</h3>
<p>How do singleton scores compare to proteins in orthogroups?</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Add clustering status to all proteins</span></span>
<span><span class="va">clustered_proteins</span> <span class="op">&lt;-</span> <span class="va">clusters</span><span class="op">$</span><span class="va">orthogroups</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">protein_id</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>status <span class="op">=</span> <span class="st">"In orthogroup"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">singleton_status</span> <span class="op">&lt;-</span> <span class="va">singletons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">protein_id</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>status <span class="op">=</span> <span class="st">"Singleton"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">protein_status</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind_rows.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="va">clustered_proteins</span>, <span class="va">singleton_status</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Join with scores</span></span>
<span><span class="va">score_comparison</span> <span class="op">&lt;-</span> <span class="va">all_proteins</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">inner_join</a></span><span class="op">(</span><span class="va">protein_status</span>, by <span class="op">=</span> <span class="st">"protein_id"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot comparison</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">score_comparison</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">custom_score</span>, fill <span class="op">=</span> <span class="va">status</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_density.html" class="external-link">geom_density</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.6</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"In orthogroup"</span> <span class="op">=</span> <span class="st">"steelblue"</span>, <span class="st">"Singleton"</span> <span class="op">=</span> <span class="st">"coral"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>    title <span class="op">=</span> <span class="st">"Effector Score Distribution: Singletons vs Orthogroup Members"</span>,</span>
<span>    x <span class="op">=</span> <span class="st">"Effector Prediction Score"</span>,</span>
<span>    y <span class="op">=</span> <span class="st">"Density"</span>,</span>
<span>    fill <span class="op">=</span> <span class="st">"Clustering Status"</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span><span class="op">)</span></span></code></pre></div>
<p><img src="effector-analysis_files/figure-html/singleton-score-distribution-1.png" class="r-plt" alt="" width="1050"></p>
<p>This shows that singletons span the full range of effector scores -
some are high-confidence effector candidates, while others may be
lower-priority.</p>
</div>
<div class="section level3">
<h3 id="exporting-singletons-for-follow-up">Exporting Singletons for Follow-up<a class="anchor" aria-label="anchor" href="#exporting-singletons-for-follow-up"></a>
</h3>
<p>Export high-scoring singletons for experimental validation:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># High-confidence singleton effectors</span></span>
<span><span class="va">high_score_singletons</span> <span class="op">&lt;-</span> <span class="va">singleton_scores</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">custom_score</span> <span class="op">&gt;=</span> <span class="fl">5.0</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">high_score_singletons</span>, <span class="st">"high_confidence_singletons.csv"</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="pan-effectorome-structure">Pan-Effectorome Structure<a class="anchor" aria-label="anchor" href="#pan-effectorome-structure"></a>
</h2>
<p>Beyond singletons, it’s useful to see the overall structure of the
effector pan-genome. paneffectR provides functions to visualize how
proteins are distributed across assemblies.</p>
<div class="section level3">
<h3 id="visualizing-pan-effectorome-structure">Visualizing Pan-Effectorome Structure<a class="anchor" aria-label="anchor" href="#visualizing-pan-effectorome-structure"></a>
</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_pan_structure.html">plot_pan_structure</a></span><span class="op">(</span><span class="va">clusters</span><span class="op">)</span></span></code></pre></div>
<p><img src="effector-analysis_files/figure-html/pan-effectorome-bar-1.png" class="r-plt" alt="" width="1050"></p>
<p>This shows the composition of the pan-effectorome:</p>
<ul>
<li>
<strong>Core</strong>: Present in all assemblies - likely essential
for pathogenicity</li>
<li>
<strong>Accessory</strong>: Present in many but not all
assemblies</li>
<li>
<strong>Rare</strong>: Present in few assemblies (default: ≤10% of
assemblies)</li>
<li>
<strong>Unique</strong>: Orthogroups with members from only 1
assembly</li>
<li>
<strong>Singleton</strong>: Proteins that couldn’t be clustered at
all - truly unique</li>
</ul>
<p>The thresholds for Rare and Accessory categories are
configurable:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Custom thresholds: Rare if in ≤20% of assemblies, Accessory if ≤50%</span></span>
<span><span class="fu"><a href="../reference/plot_pan_structure.html">plot_pan_structure</a></span><span class="op">(</span><span class="va">clusters</span>, rare_threshold <span class="op">=</span> <span class="fl">0.20</span>, accessory_threshold <span class="op">=</span> <span class="fl">0.50</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="per-assembly-breakdown">Per-Assembly Breakdown<a class="anchor" aria-label="anchor" href="#per-assembly-breakdown"></a>
</h3>
<p>How does each assembly contribute to the pan-effectorome?</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_assembly_composition.html">plot_assembly_composition</a></span><span class="op">(</span><span class="va">clusters</span><span class="op">)</span></span></code></pre></div>
<p><img src="effector-analysis_files/figure-html/per-assembly-breakdown-1.png" class="r-plt" alt="" width="1050"></p>
<p>Use <code>position = "fill"</code> to show proportions instead of
counts:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_assembly_composition.html">plot_assembly_composition</a></span><span class="op">(</span><span class="va">clusters</span>, position <span class="op">=</span> <span class="st">"fill"</span><span class="op">)</span></span></code></pre></div>
<p><img src="effector-analysis_files/figure-html/per-assembly-proportions-1.png" class="r-plt" alt="" width="1050"></p>
</div>
</div>
<div class="section level2">
<h2 id="working-with-effector-scores">Working with Effector Scores<a class="anchor" aria-label="anchor" href="#working-with-effector-scores"></a>
</h2>
<div class="section level3">
<h3 id="score-based-filtering">Score-Based Filtering<a class="anchor" aria-label="anchor" href="#score-based-filtering"></a>
</h3>
<p>When building matrices, you can filter to high-confidence
effectors:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Build unfiltered matrix first</span></span>
<span><span class="va">pa</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_pa_matrix.html">build_pa_matrix</a></span><span class="op">(</span><span class="va">clusters</span>, type <span class="op">=</span> <span class="st">"binary"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Build matrix with score threshold</span></span>
<span><span class="co"># Only orthogroups containing proteins with score &gt;= 5 are included</span></span>
<span><span class="va">pa_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_pa_matrix.html">build_pa_matrix</a></span><span class="op">(</span></span>
<span>  <span class="va">clusters</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"binary"</span>,</span>
<span>  score_threshold <span class="op">=</span> <span class="fl">5.0</span>,</span>
<span>  proteins <span class="op">=</span> <span class="va">proteins</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"All orthogroups:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">pa</span><span class="op">$</span><span class="va">matrix</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; All orthogroups: 62</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"After score filter (&gt;=5):"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">pa_filtered</span><span class="op">$</span><span class="va">matrix</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; After score filter (&gt;=5): 31</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="score-based-matrix">Score-Based Matrix<a class="anchor" aria-label="anchor" href="#score-based-matrix"></a>
</h3>
<p>Instead of binary presence/absence, create a matrix with actual
scores:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pa_score</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_pa_matrix.html">build_pa_matrix</a></span><span class="op">(</span></span>
<span>  <span class="va">clusters</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"score"</span>,</span>
<span>  score_column <span class="op">=</span> <span class="st">"custom_score"</span>,</span>
<span>  proteins <span class="op">=</span> <span class="va">proteins</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View scores (0 = absent, otherwise the score value)</span></span>
<span><span class="va">pa_score</span><span class="op">$</span><span class="va">matrix</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="op">]</span></span>
<span><span class="co">#&gt;              asm_A    asm_B    asm_C    asm_D    asm_E    asm_F</span></span>
<span><span class="co">#&gt; OG_acc_01 5.715893       NA       NA       NA 5.166307 4.349692</span></span>
<span><span class="co">#&gt; OG_acc_02 4.573136 4.449630       NA 6.105342 6.137450       NA</span></span>
<span><span class="co">#&gt; OG_acc_03 6.999290 5.138241 5.778877 6.306409 5.961777       NA</span></span>
<span><span class="co">#&gt; OG_acc_04 5.505549 6.366047 5.559538 5.049768       NA       NA</span></span>
<span><span class="co">#&gt; OG_acc_05 6.460179       NA 4.726615 6.125112 6.574661 6.382732</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="visualizing-effector-repertoires">Visualizing Effector Repertoires<a class="anchor" aria-label="anchor" href="#visualizing-effector-repertoires"></a>
</h2>
<div class="section level3">
<h3 id="heatmap">Heatmap<a class="anchor" aria-label="anchor" href="#heatmap"></a>
</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ht</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span></span>
<span>  <span class="va">pa</span>,</span>
<span>  cluster_rows <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  cluster_cols <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  show_row_names <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"white"</span>, <span class="st">"darkred"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu">ComplexHeatmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/draw-dispatch.html" class="external-link">draw</a></span><span class="op">(</span><span class="va">ht</span><span class="op">)</span></span></code></pre></div>
<p><img src="effector-analysis_files/figure-html/heatmap-1.png" class="r-plt" alt="" width="1050"></p>
</div>
<div class="section level3">
<h3 id="upset-plot">UpSet Plot<a class="anchor" aria-label="anchor" href="#upset-plot"></a>
</h3>
<p>Which effector combinations are most common?</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_upset.html">plot_upset</a></span><span class="op">(</span><span class="va">pa</span>, min_size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: `aes_string()` was deprecated in ggplot2 3.0.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use tidy evaluation idioms with `aes()`.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> See also `vignette("ggplot2-in-packages")` for more information.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">UpSetR</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue to the authors.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once per session.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use `linewidth` instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">UpSetR</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue to the authors.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once per session.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; Warning: The `size` argument of `element_line()` is deprecated as of ggplot2 3.4.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use the `linewidth` argument instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">UpSetR</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue to the authors.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once per session.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span></code></pre></div>
<p><img src="effector-analysis_files/figure-html/upset-1.png" class="r-plt" alt="" width="1050"></p>
</div>
<div class="section level3">
<h3 id="assembly-dendrogram">Assembly Dendrogram<a class="anchor" aria-label="anchor" href="#assembly-dendrogram"></a>
</h3>
<p>How similar are effector repertoires between assemblies?</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_dendro.html">plot_dendro</a></span><span class="op">(</span><span class="va">pa</span>, distance_method <span class="op">=</span> <span class="st">"jaccard"</span><span class="op">)</span></span></code></pre></div>
<p><img src="effector-analysis_files/figure-html/dendro-1.png" class="r-plt" alt="" width="1050"></p>
</div>
</div>
<div class="section level2">
<h2 id="exporting-results">Exporting Results<a class="anchor" aria-label="anchor" href="#exporting-results"></a>
</h2>
<div class="section level3">
<h3 id="export-matrices">Export Matrices<a class="anchor" aria-label="anchor" href="#export-matrices"></a>
</h3>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Export presence/absence matrix</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">pa</span><span class="op">)</span>, <span class="st">"effector_presence_absence.csv"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Export orthogroup membership</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">clusters</span><span class="op">$</span><span class="va">orthogroups</span>, <span class="st">"orthogroup_membership.csv"</span>, row_names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="export-singletons">Export Singletons<a class="anchor" aria-label="anchor" href="#export-singletons"></a>
</h3>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># All singletons with scores</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">singleton_scores</span>, <span class="st">"all_singletons_with_scores.csv"</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Summary by assembly</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="../reference/singletons_by_assembly.html">singletons_by_assembly</a></span><span class="op">(</span><span class="va">clusters</span>, <span class="va">proteins</span><span class="op">)</span>,</span>
<span>  <span class="st">"singletons_summary.csv"</span>,</span>
<span>  row.names <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<p>Key steps for effector analysis with paneffectR:</p>
<ol style="list-style-type: decimal">
<li>
<strong>Load data</strong> with
<code>load_proteins(fasta_dir, score_dir)</code>
</li>
<li>
<strong>Cluster</strong> with <code><a href="../reference/cluster_proteins.html">cluster_proteins()</a></code>
(requires DIAMOND)</li>
<li>
<strong>Examine singletons first</strong> - use
<code><a href="../reference/get_singletons.html">get_singletons()</a></code> and check their scores</li>
<li>
<strong>Build matrices</strong> with <code><a href="../reference/build_pa_matrix.html">build_pa_matrix()</a></code>,
optionally filtering by score</li>
<li>
<strong>Visualize</strong> with <code><a href="../reference/plot_heatmap.html">plot_heatmap()</a></code>,
<code><a href="../reference/plot_upset.html">plot_upset()</a></code>, <code><a href="../reference/plot_dendro.html">plot_dendro()</a></code>
</li>
<li>
<strong>Export</strong> singletons and orthogroup membership for
follow-up</li>
</ol>
</div>
<div class="section level2">
<h2 id="next-steps">Next Steps<a class="anchor" aria-label="anchor" href="#next-steps"></a>
</h2>
<ul>
<li>
<strong><a href="getting-started.html">Getting Started</a></strong>
- Basic workflow overview</li>
<li>
<strong><a href="pan-genome.html">Pan-Genome Analysis</a></strong> -
Analysis without scores</li>
<li>
<strong><a href="algorithms.html">Algorithm Deep Dive</a></strong> -
Technical details on clustering</li>
</ul>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://github.com/danmaclean" class="external-link">Dan MacLean</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
