<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Effector Analysis with paneffectR • paneffectR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Effector Analysis with paneffectR">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">paneffectR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes">
<li><a class="dropdown-item" href="../articles/getting-started.html">Getting Started</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Workflows</h6></li>
    <li><a class="dropdown-item" href="../articles/effector-analysis.html">Effector Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/pan-genome.html">Pan-Genome Analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Technical</h6></li>
    <li><a class="dropdown-item" href="../articles/algorithms.html">Algorithm Deep Dive</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/TeamMacLean/paneffectR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Effector Analysis with paneffectR</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/TeamMacLean/paneffectR/blob/main/vignettes/effector-analysis.Rmd" class="external-link"><code>vignettes/effector-analysis.Rmd</code></a></small>
      <div class="d-none name"><code>effector-analysis.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette demonstrates how to use paneffectR with output from the
<a href="https://github.com/TeamMacLean/omnieff" class="external-link">omnieff</a> effector
prediction pipeline. The key advantage is the ability to <strong>filter
by effector prediction scores</strong>, focusing your analysis on
high-confidence effector candidates.</p>
</div>
<div class="section level2">
<h2 id="the-omnieff-pipeline">The omnieff Pipeline<a class="anchor" aria-label="anchor" href="#the-omnieff-pipeline"></a>
</h2>
<p>omnieff runs multiple effector prediction tools and combines their
results:</p>
<ul>
<li>
<strong>Signal peptide predictors</strong>: SignalP 3/4/5/6,
Phobius</li>
<li>
<strong>Effector predictors</strong>: EffectorP 1/2/3,
DeepRedEff</li>
<li>
<strong>Localization</strong>: ApoplastP, Localizer</li>
<li>
<strong>Domain annotations</strong>: Pfam</li>
</ul>
<p>The output includes:</p>
<ol style="list-style-type: decimal">
<li>
<strong>FASTA files</strong> (<code>*_reformatted.faa</code>):
Protein sequences with standardized names</li>
<li>
<strong>Score files</strong> (<code>*_scored.csv</code>): All
prediction scores plus a combined <code>custom_score</code>
</li>
<li>
<strong>Name mapping</strong> (<code>*_name_mapping.csv</code>):
Links new names to original contig names</li>
</ol>
</div>
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/TeamMacLean/paneffectR" class="external-link">paneffectR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="loading-omnieff-output">Loading omnieff Output<a class="anchor" aria-label="anchor" href="#loading-omnieff-output"></a>
</h2>
<p>Use <code><a href="../reference/load_proteins.html">load_proteins()</a></code> with both FASTA and score
directories:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get path to test data (simulates omnieff output structure)</span></span>
<span><span class="va">testdata_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"testdata"</span>, package <span class="op">=</span> <span class="st">"paneffectR"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load with scores</span></span>
<span><span class="va">proteins</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_proteins.html">load_proteins</a></span><span class="op">(</span></span>
<span>  fasta_dir <span class="op">=</span> <span class="va">testdata_dir</span>,</span>
<span>  score_dir <span class="op">=</span> <span class="va">testdata_dir</span>,</span>
<span>  pattern <span class="op">=</span> <span class="st">"*.faa"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">proteins</span></span>
<span><span class="co">#&gt; -- protein_collection --</span></span>
<span><span class="co">#&gt; 3 assemblies, 300 total proteins</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 3</span></span></span>
<span><span class="co">#&gt;   assembly_name n_proteins has_scores</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> assembly1            100 TRUE      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> assembly2            100 TRUE      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> assembly3            100 TRUE</span></span></code></pre></div>
<p>Notice that <code>has_scores</code> is TRUE for all assemblies.</p>
</div>
<div class="section level2">
<h2 id="understanding-score-columns">Understanding Score Columns<a class="anchor" aria-label="anchor" href="#understanding-score-columns"></a>
</h2>
<p>The score data contains many columns from different prediction
tools:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get one assembly to explore</span></span>
<span><span class="va">ps</span> <span class="op">&lt;-</span> <span class="va">proteins</span><span class="op">$</span><span class="va">assemblies</span><span class="op">[[</span><span class="st">"assembly1"</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># All available columns</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ps</span><span class="op">$</span><span class="va">proteins</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "protein_id"                       "sequence"                        </span></span>
<span><span class="co">#&gt;  [3] "assembly"                         "signalp3_nn"                     </span></span>
<span><span class="co">#&gt;  [5] "signalp3_nn_d"                    "signalp3_hmm"                    </span></span>
<span><span class="co">#&gt;  [7] "signalp3_hmm_s"                   "signalp4"                        </span></span>
<span><span class="co">#&gt;  [9] "signalp4_d"                       "signalp5"                        </span></span>
<span><span class="co">#&gt; [11] "signalp5_d"                       "signalp5_prediction"             </span></span>
<span><span class="co">#&gt; [13] "phobius_tm_count"                 "phobius_has_sp"                  </span></span>
<span><span class="co">#&gt; [15] "phobius_prediction"               "effectorp1"                      </span></span>
<span><span class="co">#&gt; [17] "effectorp1_prob"                  "effectorp2"                      </span></span>
<span><span class="co">#&gt; [19] "effectorp2_prob"                  "effectorp3_cytoplasmic_fungal"   </span></span>
<span><span class="co">#&gt; [21] "effectorp3_apoplastic_fungal"     "effectorp3_noneffector_fungal"   </span></span>
<span><span class="co">#&gt; [23] "effectorp3_prediction_fungal"     "effectorp3_cytoplasmic_nonfungal"</span></span>
<span><span class="co">#&gt; [25] "effectorp3_apoplastic_nonfungal"  "effectorp3_noneffector_nonfungal"</span></span>
<span><span class="co">#&gt; [27] "effectorp3_prediction_nonfungal"  "deepredeff_fungi"                </span></span>
<span><span class="co">#&gt; [29] "deepredeff_oomycete"              "apoplastp"                       </span></span>
<span><span class="co">#&gt; [31] "apoplastp_prediction"             "tmhmm_tm_count"                  </span></span>
<span><span class="co">#&gt; [33] "tmhmm_prediction"                 "localizer"                       </span></span>
<span><span class="co">#&gt; [35] "pfam_ids"                         "pfam_names"                      </span></span>
<span><span class="co">#&gt; [37] "pfam_best_evalues"                "has_selected_pfam_match"         </span></span>
<span><span class="co">#&gt; [39] "signalp6"                         "signalp6_d"                      </span></span>
<span><span class="co">#&gt; [41] "signalp6_prediction"              "custom_score"                    </span></span>
<span><span class="co">#&gt; [43] "score_rank"</span></span></code></pre></div>
<div class="section level3">
<h3 id="key-score-columns">Key Score Columns<a class="anchor" aria-label="anchor" href="#key-score-columns"></a>
</h3>
<table class="table">
<colgroup>
<col width="38%">
<col width="61%">
</colgroup>
<thead><tr class="header">
<th>Column</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>custom_score</code></td>
<td>Combined effector score (higher = more likely effector)</td>
</tr>
<tr class="even">
<td><code>score_rank</code></td>
<td>Rank by custom_score (1 = highest)</td>
</tr>
<tr class="odd">
<td><code>effectorp1_prob</code></td>
<td>EffectorP v1 probability</td>
</tr>
<tr class="even">
<td><code>effectorp2_prob</code></td>
<td>EffectorP v2 probability</td>
</tr>
<tr class="odd">
<td><code>effectorp3_prediction_fungal</code></td>
<td>EffectorP v3 category (fungal mode)</td>
</tr>
<tr class="even">
<td><code>deepredeff_fungi</code></td>
<td>DeepRedEff fungal score</td>
</tr>
<tr class="odd">
<td><code>signalp5_d</code></td>
<td>SignalP5 D-score</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="exploring-score-distributions">Exploring Score Distributions<a class="anchor" aria-label="anchor" href="#exploring-score-distributions"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Top-scoring proteins in assembly1</span></span>
<span><span class="va">ps</span><span class="op">$</span><span class="va">proteins</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">score_rank</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">protein_id</span>, <span class="va">custom_score</span>, <span class="va">score_rank</span>,</span>
<span>         <span class="va">effectorp2_prob</span>, <span class="va">signalp5_d</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 × 5</span></span></span>
<span><span class="co">#&gt;    protein_id       custom_score score_rank effectorp2_prob signalp5_d</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> assembly1_000267         5.92          1           0.764    0.062<span style="text-decoration: underline;">3</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> assembly1_003502         5.14          2           0.613    0.352  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> assembly1_001347         5.06          3           0.996    0.181  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> assembly1_005840         5.06          4           0.995    0.446  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> assembly1_006291         5.00          5           0.999    0.009<span style="text-decoration: underline;">65</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> assembly1_004421         4.99          6           0.982    0.075<span style="text-decoration: underline;">9</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> assembly1_001504         4.99          7           0.979    0.829  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> assembly1_002717         4.95          8           0.974    0.039<span style="text-decoration: underline;">8</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> assembly1_002883         4.93          9           0.969    0.029<span style="text-decoration: underline;">3</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> assembly1_004234         4.91         10           0.998    0.182</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="visualizing-score-distributions">Visualizing Score Distributions<a class="anchor" aria-label="anchor" href="#visualizing-score-distributions"></a>
</h2>
<p>Before filtering, understand your score distributions:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Overall distribution</span></span>
<span><span class="fu"><a href="../reference/plot_scores.html">plot_scores</a></span><span class="op">(</span><span class="va">proteins</span><span class="op">)</span></span></code></pre></div>
<p><img src="effector-analysis_files/figure-html/score-plots-1.png" class="r-plt" alt="" width="1050"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># By assembly</span></span>
<span><span class="fu"><a href="../reference/plot_scores.html">plot_scores</a></span><span class="op">(</span><span class="va">proteins</span>, by_assembly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="effector-analysis_files/figure-html/score-plots-2.png" class="r-plt" alt="" width="1050"></p>
<div class="section level3">
<h3 id="choosing-a-threshold">Choosing a Threshold<a class="anchor" aria-label="anchor" href="#choosing-a-threshold"></a>
</h3>
<p>The <code>custom_score</code> combines multiple predictors. Higher
scores indicate stronger effector evidence. Use a threshold to focus on
high-confidence candidates:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Show threshold line at score = 5</span></span>
<span><span class="fu"><a href="../reference/plot_scores.html">plot_scores</a></span><span class="op">(</span><span class="va">proteins</span>, threshold <span class="op">=</span> <span class="fl">5.0</span><span class="op">)</span></span></code></pre></div>
<p><img src="effector-analysis_files/figure-html/score-threshold-viz-1.png" class="r-plt" alt="" width="1050"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Histogram view</span></span>
<span><span class="fu"><a href="../reference/plot_scores.html">plot_scores</a></span><span class="op">(</span><span class="va">proteins</span>, plot_type <span class="op">=</span> <span class="st">"histogram"</span>, threshold <span class="op">=</span> <span class="fl">5.0</span><span class="op">)</span></span></code></pre></div>
<p><img src="effector-analysis_files/figure-html/score-threshold-viz-2.png" class="r-plt" alt="" width="1050"></p>
</div>
</div>
<div class="section level2">
<h2 id="clustering-and-filtering">Clustering and Filtering<a class="anchor" aria-label="anchor" href="#clustering-and-filtering"></a>
</h2>
<div class="section level3">
<h3 id="basic-clustering">Basic Clustering<a class="anchor" aria-label="anchor" href="#basic-clustering"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Cluster all proteins (requires DIAMOND)</span></span>
<span><span class="va">clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cluster_proteins.html">cluster_proteins</a></span><span class="op">(</span><span class="va">proteins</span>, method <span class="op">=</span> <span class="st">"diamond_rbh"</span><span class="op">)</span></span></code></pre></div>
<p>For this vignette, we’ll use pre-computed clusters:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">visual_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"testdata"</span>, <span class="st">"visual"</span>, package <span class="op">=</span> <span class="st">"paneffectR"</span><span class="op">)</span></span>
<span><span class="va">clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">visual_dir</span>, <span class="st">"clusters_visual.rds"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="building-a-filtered-matrix">Building a Filtered Matrix<a class="anchor" aria-label="anchor" href="#building-a-filtered-matrix"></a>
</h3>
<p>Filter to only high-confidence effectors when building the
matrix:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the proteins that match the clusters</span></span>
<span><span class="va">proteins_visual</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">visual_dir</span>, <span class="st">"proteins_visual.rds"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Build matrix with score threshold</span></span>
<span><span class="co"># Only orthogroups containing proteins with score &gt;= 3 are included</span></span>
<span><span class="va">pa_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_pa_matrix.html">build_pa_matrix</a></span><span class="op">(</span></span>
<span>  <span class="va">clusters</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"binary"</span>,</span>
<span>  score_threshold <span class="op">=</span> <span class="fl">3.0</span>,</span>
<span>  proteins <span class="op">=</span> <span class="va">proteins_visual</span></span>
<span><span class="op">)</span></span>
<span><span class="va">pa_filtered</span></span>
<span><span class="co">#&gt; -- pa_matrix (binary) --</span></span>
<span><span class="co">#&gt; 36 orthogroups x 6 assemblies</span></span>
<span><span class="co">#&gt; Sparsity: 34.3%</span></span>
<span><span class="co">#&gt; Threshold: 3</span></span></code></pre></div>
<p>Compare with unfiltered:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pa_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_pa_matrix.html">build_pa_matrix</a></span><span class="op">(</span><span class="va">clusters</span>, type <span class="op">=</span> <span class="st">"binary"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"All orthogroups:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">pa_all</span><span class="op">$</span><span class="va">matrix</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; All orthogroups: 50</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"After score filter:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">pa_filtered</span><span class="op">$</span><span class="va">matrix</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; After score filter: 36</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="score-based-matrix">Score-Based Matrix<a class="anchor" aria-label="anchor" href="#score-based-matrix"></a>
</h2>
<p>Instead of binary presence/absence, you can create a matrix with
actual scores:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Use maximum score per orthogroup per assembly</span></span>
<span><span class="va">pa_score</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_pa_matrix.html">build_pa_matrix</a></span><span class="op">(</span></span>
<span>  <span class="va">clusters</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"score"</span>,</span>
<span>  score_column <span class="op">=</span> <span class="st">"custom_score"</span>,</span>
<span>  proteins <span class="op">=</span> <span class="va">proteins_visual</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View scores (0 = absent, otherwise the score value)</span></span>
<span><span class="va">pa_score</span><span class="op">$</span><span class="va">matrix</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="op">]</span></span>
<span><span class="co">#&gt;              asm_A    asm_B    asm_C    asm_D    asm_E    asm_F</span></span>
<span><span class="co">#&gt; OG_acc_01 6.329748       NA       NA       NA 5.634757 4.490995</span></span>
<span><span class="co">#&gt; OG_acc_02 4.880568 4.297836       NA 6.800552 6.062126       NA</span></span>
<span><span class="co">#&gt; OG_acc_03 5.043444 4.251004 6.456519 5.171333 6.114216       NA</span></span>
<span><span class="co">#&gt; OG_acc_04 5.947353 6.840645 6.917769 4.371568       NA       NA</span></span>
<span><span class="co">#&gt; OG_acc_05 6.531186       NA 4.749513 5.804187 6.686394 5.975107</span></span></code></pre></div>
<p>This is useful when you want to see not just presence/absence, but
how strongly each orthogroup is predicted as an effector in each
assembly.</p>
</div>
<div class="section level2">
<h2 id="visualizing-effector-repertoires">Visualizing Effector Repertoires<a class="anchor" aria-label="anchor" href="#visualizing-effector-repertoires"></a>
</h2>
<div class="section level3">
<h3 id="heatmap-with-score-colors">Heatmap with Score Colors<a class="anchor" aria-label="anchor" href="#heatmap-with-score-colors"></a>
</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Binary heatmap (presence/absence)</span></span>
<span><span class="va">ht</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span></span>
<span>  <span class="va">pa_filtered</span>,</span>
<span>  cluster_rows <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  cluster_cols <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  show_row_names <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"white"</span>, <span class="st">"darkred"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu">ComplexHeatmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/draw-dispatch.html" class="external-link">draw</a></span><span class="op">(</span><span class="va">ht</span><span class="op">)</span></span></code></pre></div>
<p><img src="effector-analysis_files/figure-html/heatmap-scores-1.png" class="r-plt" alt="" width="1050"></p>
</div>
<div class="section level3">
<h3 id="upset-plot-for-effector-sharing">UpSet Plot for Effector Sharing<a class="anchor" aria-label="anchor" href="#upset-plot-for-effector-sharing"></a>
</h3>
<p>Which effectors are shared vs unique?</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_upset.html">plot_upset</a></span><span class="op">(</span><span class="va">pa_filtered</span>, min_size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: `aes_string()` was deprecated in ggplot2 3.0.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use tidy evaluation idioms with `aes()`.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> See also `vignette("ggplot2-in-packages")` for more information.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">UpSetR</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue to the authors.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once per session.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use `linewidth` instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">UpSetR</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue to the authors.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once per session.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; Warning: The `size` argument of `element_line()` is deprecated as of ggplot2 3.4.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use the `linewidth` argument instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">UpSetR</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue to the authors.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once per session.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span></code></pre></div>
<p><img src="effector-analysis_files/figure-html/upset-effectors-1.png" class="r-plt" alt="" width="1050"></p>
</div>
<div class="section level3">
<h3 id="assembly-similarity">Assembly Similarity<a class="anchor" aria-label="anchor" href="#assembly-similarity"></a>
</h3>
<p>How similar are the effector repertoires between assemblies?</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_dendro.html">plot_dendro</a></span><span class="op">(</span><span class="va">pa_filtered</span>, distance_method <span class="op">=</span> <span class="st">"jaccard"</span><span class="op">)</span></span></code></pre></div>
<p><img src="effector-analysis_files/figure-html/dendro-effectors-1.png" class="r-plt" alt="" width="1050"></p>
</div>
</div>
<div class="section level2">
<h2 id="identifying-core-vs-accessory-effectors">Identifying Core vs Accessory Effectors<a class="anchor" aria-label="anchor" href="#identifying-core-vs-accessory-effectors"></a>
</h2>
<div class="section level3">
<h3 id="core-effectors">Core Effectors<a class="anchor" aria-label="anchor" href="#core-effectors"></a>
</h3>
<p>Present in all assemblies - likely essential for pathogenicity:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Find orthogroups present in all assemblies</span></span>
<span><span class="va">n_assemblies</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">pa_filtered</span><span class="op">$</span><span class="va">matrix</span><span class="op">)</span></span>
<span><span class="va">presence_count</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">pa_filtered</span><span class="op">$</span><span class="va">matrix</span><span class="op">)</span></span>
<span><span class="va">core_og</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">presence_count</span><span class="op">[</span><span class="va">presence_count</span> <span class="op">==</span> <span class="va">n_assemblies</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Core effector orthogroups:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">core_og</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Core effector orthogroups: 10</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Names:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">core_og</span>, collapse <span class="op">=</span> <span class="st">", "</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Names: OG_core_01, OG_core_02, OG_core_03, OG_core_04, OG_core_05, OG_core_06, OG_core_07, OG_core_08, OG_core_09, OG_core_10</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="accessory-effectors">Accessory Effectors<a class="anchor" aria-label="anchor" href="#accessory-effectors"></a>
</h3>
<p>Present in some but not all assemblies:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">accessory_og</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">presence_count</span><span class="op">[</span><span class="va">presence_count</span> <span class="op">&gt;</span> <span class="fl">1</span> <span class="op">&amp;</span> <span class="va">presence_count</span> <span class="op">&lt;</span> <span class="va">n_assemblies</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Accessory effector orthogroups:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">accessory_og</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Accessory effector orthogroups: 26</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="unique-effectors-singletons">Unique Effectors (Singletons)<a class="anchor" aria-label="anchor" href="#unique-effectors-singletons"></a>
</h3>
<p>Present in only one assembly - may contribute to host
specificity:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">unique_og</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">presence_count</span><span class="op">[</span><span class="va">presence_count</span> <span class="op">==</span> <span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Unique effector orthogroups:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">unique_og</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Unique effector orthogroups: 0</span></span>
<span></span>
<span><span class="co"># Which assemblies have them?</span></span>
<span><span class="va">unique_assemblies</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">pa_filtered</span><span class="op">$</span><span class="va">matrix</span><span class="op">[</span><span class="va">unique_og</span>, , drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">[</span><span class="va">x</span> <span class="op">==</span> <span class="fl">1</span><span class="op">]</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">unique_assemblies</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; &lt; table of extent 0 &gt;</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="exporting-results">Exporting Results<a class="anchor" aria-label="anchor" href="#exporting-results"></a>
</h2>
<div class="section level3">
<h3 id="export-to-csv">Export to CSV<a class="anchor" aria-label="anchor" href="#export-to-csv"></a>
</h3>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Export presence/absence matrix</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">pa_filtered</span><span class="op">)</span>,</span>
<span>  <span class="st">"effector_presence_absence.csv"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Export orthogroup membership</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span></span>
<span>  <span class="va">clusters</span><span class="op">$</span><span class="va">orthogroups</span>,</span>
<span>  <span class="st">"orthogroup_membership.csv"</span>,</span>
<span>  row.names <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="get-proteins-from-specific-orthogroups">Get Proteins from Specific Orthogroups<a class="anchor" aria-label="anchor" href="#get-proteins-from-specific-orthogroups"></a>
</h3>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get proteins from core orthogroups</span></span>
<span><span class="va">core_proteins</span> <span class="op">&lt;-</span> <span class="va">clusters</span><span class="op">$</span><span class="va">orthogroups</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">orthogroup_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">core_og</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">core_proteins</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 3</span></span></span>
<span><span class="co">#&gt;   orthogroup_id assembly protein_id</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> OG_core_01    asm_A    asm_A_p001</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> OG_core_01    asm_B    asm_B_p002</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> OG_core_01    asm_C    asm_C_p003</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> OG_core_01    asm_D    asm_D_p004</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> OG_core_01    asm_E    asm_E_p005</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> OG_core_01    asm_F    asm_F_p006</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<p>Key steps for effector analysis:</p>
<ol style="list-style-type: decimal">
<li>Load omnieff output with
<code>load_proteins(fasta_dir, score_dir)</code>
</li>
<li>Explore scores with <code><a href="../reference/plot_scores.html">plot_scores()</a></code>
</li>
<li>Cluster with <code><a href="../reference/cluster_proteins.html">cluster_proteins()</a></code>
</li>
<li>Filter by score in
<code>build_pa_matrix(score_threshold = ...)</code>
</li>
<li>Visualize with <code><a href="../reference/plot_heatmap.html">plot_heatmap()</a></code>,
<code><a href="../reference/plot_upset.html">plot_upset()</a></code>, <code><a href="../reference/plot_dendro.html">plot_dendro()</a></code>
</li>
<li>Identify core/accessory/unique effectors</li>
</ol>
</div>
<div class="section level2">
<h2 id="next-steps">Next Steps<a class="anchor" aria-label="anchor" href="#next-steps"></a>
</h2>
<ul>
<li>
<strong><a href="getting-started.html">Getting Started</a></strong>
- Basic workflow overview</li>
<li>
<strong><a href="pan-genome.html">Pan-Genome Analysis</a></strong> -
Analysis without scores</li>
<li>
<strong><a href="algorithms.html">Algorithm Deep Dive</a></strong> -
Technical details on clustering</li>
</ul>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://github.com/danmaclean" class="external-link">Dan MacLean</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
