<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Getting Started with paneffectR • paneffectR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Getting Started with paneffectR">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">paneffectR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes">
<li><a class="dropdown-item" href="../articles/getting-started.html">Getting Started</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Workflows</h6></li>
    <li><a class="dropdown-item" href="../articles/effector-analysis.html">Effector Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/pan-genome.html">Pan-Genome Analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Technical</h6></li>
    <li><a class="dropdown-item" href="../articles/algorithms.html">Algorithm Deep Dive</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/TeamMacLean/paneffectR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Getting Started with paneffectR</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/TeamMacLean/paneffectR/blob/main/vignettes/getting-started.Rmd" class="external-link"><code>vignettes/getting-started.Rmd</code></a></small>
      <div class="d-none name"><code>getting-started.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>paneffectR helps you compare proteins across multiple genome
assemblies. The typical workflow is: 1. <strong>Load</strong> protein
sequences from FASTA files 2. <strong>Cluster</strong> proteins into
orthogroups (groups of equivalent proteins) 3. <strong>Build</strong> a
presence/absence matrix 4. <strong>Visualize</strong> the results</p>
<p>This vignette walks through each step using example data included
with the package.</p>
</div>
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/TeamMacLean/paneffectR" class="external-link">paneffectR</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="step-1-load-protein-data">Step 1: Load Protein Data<a class="anchor" aria-label="anchor" href="#step-1-load-protein-data"></a>
</h2>
<p>The main entry point is <code><a href="../reference/load_proteins.html">load_proteins()</a></code>, which discovers
and loads all FASTA files from a directory:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get path to test data included with the package</span></span>
<span><span class="va">testdata_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"testdata"</span>, package <span class="op">=</span> <span class="st">"paneffectR"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load all assemblies (FASTAs and scores)</span></span>
<span><span class="va">proteins</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_proteins.html">load_proteins</a></span><span class="op">(</span></span>
<span>  fasta_dir <span class="op">=</span> <span class="va">testdata_dir</span>,</span>
<span>  score_dir <span class="op">=</span> <span class="va">testdata_dir</span>,</span>
<span>  pattern <span class="op">=</span> <span class="st">"*.faa"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">proteins</span></span>
<span><span class="co">#&gt; -- protein_collection --</span></span>
<span><span class="co">#&gt; 3 assemblies, 300 total proteins</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 3</span></span></span>
<span><span class="co">#&gt;   assembly_name n_proteins has_scores</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> assembly1            100 TRUE      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> assembly2            100 TRUE      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> assembly3            100 TRUE</span></span></code></pre></div>
<p>The result is a <code>protein_collection</code> containing all
assemblies. You can see:</p>
<ul>
<li>How many assemblies were loaded</li>
<li>How many proteins in each</li>
<li>Whether effector scores are present</li>
</ul>
<div class="section level3">
<h3 id="exploring-the-data">Exploring the Data<a class="anchor" aria-label="anchor" href="#exploring-the-data"></a>
</h3>
<p>Each assembly is stored as a <code>protein_set</code>. Access
individual assemblies by name:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Access one assembly</span></span>
<span><span class="va">ps</span> <span class="op">&lt;-</span> <span class="va">proteins</span><span class="op">$</span><span class="va">assemblies</span><span class="op">[[</span><span class="st">"assembly1"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">ps</span></span>
<span><span class="co">#&gt; -- protein_set: assembly1 --</span></span>
<span><span class="co">#&gt; 100 proteins</span></span>
<span><span class="co">#&gt; Scores: present</span></span>
<span></span>
<span><span class="co"># The proteins are stored as a tibble</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ps</span><span class="op">$</span><span class="va">proteins</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"protein_id"</span>, <span class="st">"sequence"</span>, <span class="st">"custom_score"</span>, <span class="st">"score_rank"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 4</span></span></span>
<span><span class="co">#&gt;   protein_id       sequence                              custom_score score_rank</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> assembly1_000039 MPFVLNLLVLVAVWYGLVRMAMLLKDDSEIKSFDPF…         4.62         93</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> assembly1_000040 MEVAECTHKLASTTFVKTVEAPALAIMEPLLKRALA…         4.75         64</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> assembly1_000094 MTDKDKPKDDDDCYVKVNTPGGVGDTNSPNKGGLPF…         4.86         27</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> assembly1_000122 MGSSPIFRPFKCKFSMFSKSILIFLLILPLISVILL…         4.81         44</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> assembly1_000164 MLFCVMVGTGCQLLGMALVTLFFAAVGVLAPSNRGK…         4.78         54</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> assembly1_000194 MMGNLAKDIVNGHREKTLALLWKLISCFQLQALVDA…         4.76         59</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="step-2-cluster-proteins">Step 2: Cluster Proteins<a class="anchor" aria-label="anchor" href="#step-2-cluster-proteins"></a>
</h2>
<p>Now we group proteins from different assemblies into
<strong>orthogroups</strong> - sets of proteins that are likely the same
gene/function across assemblies.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Cluster using DIAMOND reciprocal best hits</span></span>
<span><span class="va">clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cluster_proteins.html">cluster_proteins</a></span><span class="op">(</span><span class="va">proteins</span>, method <span class="op">=</span> <span class="st">"diamond_rbh"</span><span class="op">)</span></span>
<span><span class="va">clusters</span></span></code></pre></div>
<p><strong>Note:</strong> Clustering requires DIAMOND to be installed.
If you don’t have DIAMOND, you can skip this step and use the
pre-computed clusters included with the package for the visualization
examples.</p>
<p>For this vignette, we’ll load pre-computed results:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load pre-computed visual test data</span></span>
<span><span class="va">visual_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"testdata"</span>, <span class="st">"visual"</span>, package <span class="op">=</span> <span class="st">"paneffectR"</span><span class="op">)</span></span>
<span><span class="va">clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">visual_dir</span>, <span class="st">"clusters_visual.rds"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">clusters</span></span>
<span><span class="co">#&gt; -- orthogroup_result (synthetic_visual) --</span></span>
<span><span class="co">#&gt; 50 orthogroups</span></span></code></pre></div>
<div class="section level3">
<h3 id="understanding-orthogroups">Understanding Orthogroups<a class="anchor" aria-label="anchor" href="#understanding-orthogroups"></a>
</h3>
<p>An orthogroup result contains:</p>
<ul>
<li>
<strong>orthogroups</strong>: Which proteins belong to which
group</li>
<li>
<strong>singletons</strong>: Proteins not assigned to any group
(unique to one assembly)</li>
<li>
<strong>stats</strong>: Summary statistics</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># See the orthogroup assignments</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">clusters</span><span class="op">$</span><span class="va">orthogroups</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 3</span></span></span>
<span><span class="co">#&gt;   orthogroup_id assembly protein_id</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> OG_core_01    asm_A    asm_A_p001</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> OG_core_01    asm_B    asm_B_p002</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> OG_core_01    asm_C    asm_C_p003</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> OG_core_01    asm_D    asm_D_p004</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> OG_core_01    asm_E    asm_E_p005</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> OG_core_01    asm_F    asm_F_p006</span></span>
<span></span>
<span><span class="co"># How many singletons?</span></span>
<span><span class="fu"><a href="../reference/n_singletons.html">n_singletons</a></span><span class="op">(</span><span class="va">clusters</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0</span></span>
<span></span>
<span><span class="co"># Singletons by assembly</span></span>
<span><span class="fu"><a href="../reference/singletons_by_assembly.html">singletons_by_assembly</a></span><span class="op">(</span><span class="va">clusters</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 0 × 2</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 2 variables: assembly &lt;chr&gt;, n_singletons &lt;int&gt;</span></span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="step-3-build-presenceabsence-matrix">Step 3: Build Presence/Absence Matrix<a class="anchor" aria-label="anchor" href="#step-3-build-presenceabsence-matrix"></a>
</h2>
<p>Convert the orthogroups into a matrix where:</p>
<ul>
<li>
<strong>Rows</strong> = orthogroups</li>
<li>
<strong>Columns</strong> = assemblies</li>
<li>
<strong>Values</strong> = presence (1) or absence (0)</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Build binary presence/absence matrix</span></span>
<span><span class="va">pa</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_pa_matrix.html">build_pa_matrix</a></span><span class="op">(</span><span class="va">clusters</span>, type <span class="op">=</span> <span class="st">"binary"</span><span class="op">)</span></span>
<span><span class="va">pa</span></span>
<span><span class="co">#&gt; -- pa_matrix (binary) --</span></span>
<span><span class="co">#&gt; 50 orthogroups x 6 assemblies</span></span>
<span><span class="co">#&gt; Sparsity: 46.7%</span></span>
<span></span>
<span><span class="co"># View the raw matrix</span></span>
<span><span class="va">pa</span><span class="op">$</span><span class="va">matrix</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span></span>
<span><span class="co">#&gt;           asm_A asm_B asm_C asm_D asm_E asm_F</span></span>
<span><span class="co">#&gt; OG_acc_01     1     0     0     0     1     1</span></span>
<span><span class="co">#&gt; OG_acc_02     1     1     0     1     1     0</span></span>
<span><span class="co">#&gt; OG_acc_03     1     1     1     1     1     0</span></span>
<span><span class="co">#&gt; OG_acc_04     1     1     1     1     0     0</span></span>
<span><span class="co">#&gt; OG_acc_05     1     0     1     1     1     1</span></span>
<span><span class="co">#&gt; OG_acc_06     1     1     1     1     0     0</span></span>
<span><span class="co">#&gt; OG_acc_07     0     1     1     0     1     1</span></span>
<span><span class="co">#&gt; OG_acc_08     0     1     0     1     1     1</span></span>
<span><span class="co">#&gt; OG_acc_09     0     0     0     1     1     1</span></span>
<span><span class="co">#&gt; OG_acc_10     0     1     1     1     1     0</span></span></code></pre></div>
<div class="section level3">
<h3 id="including-singletons">Including Singletons<a class="anchor" aria-label="anchor" href="#including-singletons"></a>
</h3>
<p>By default, singletons (proteins unique to one assembly) are included
as their own orthogroups:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Without singletons</span></span>
<span><span class="va">pa_no_single</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_pa_matrix.html">build_pa_matrix</a></span><span class="op">(</span><span class="va">clusters</span>, type <span class="op">=</span> <span class="st">"binary"</span>, exclude_singletons <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compare dimensions</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"With singletons:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">pa</span><span class="op">$</span><span class="va">matrix</span><span class="op">)</span>, <span class="st">"orthogroups\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; With singletons: 50 orthogroups</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Without singletons:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">pa_no_single</span><span class="op">$</span><span class="va">matrix</span><span class="op">)</span>, <span class="st">"orthogroups\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Without singletons: 50 orthogroups</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="step-4-visualize">Step 4: Visualize<a class="anchor" aria-label="anchor" href="#step-4-visualize"></a>
</h2>
<div class="section level3">
<h3 id="heatmap">Heatmap<a class="anchor" aria-label="anchor" href="#heatmap"></a>
</h3>
<p>The heatmap shows presence/absence patterns across all
assemblies:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ht</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">pa</span><span class="op">)</span></span>
<span><span class="fu">ComplexHeatmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/draw-dispatch.html" class="external-link">draw</a></span><span class="op">(</span><span class="va">ht</span><span class="op">)</span></span></code></pre></div>
<p><img src="getting-started_files/figure-html/heatmap-1.png" class="r-plt" alt="" width="1050"></p>
<p>Customize with clustering and colors:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ht</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span></span>
<span>  <span class="va">pa</span>,</span>
<span>  cluster_rows <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  cluster_cols <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  distance_method <span class="op">=</span> <span class="st">"jaccard"</span>,</span>
<span>  color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"white"</span>, <span class="st">"steelblue"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu">ComplexHeatmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/draw-dispatch.html" class="external-link">draw</a></span><span class="op">(</span><span class="va">ht</span><span class="op">)</span></span></code></pre></div>
<p><img src="getting-started_files/figure-html/heatmap-options-1.png" class="r-plt" alt="" width="1050"></p>
</div>
<div class="section level3">
<h3 id="upset-plot">UpSet Plot<a class="anchor" aria-label="anchor" href="#upset-plot"></a>
</h3>
<p>UpSet plots show which orthogroups are shared between assemblies:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Show intersections with at least 2 orthogroups</span></span>
<span><span class="fu"><a href="../reference/plot_upset.html">plot_upset</a></span><span class="op">(</span><span class="va">pa</span>, min_size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: `aes_string()` was deprecated in ggplot2 3.0.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use tidy evaluation idioms with `aes()`.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> See also `vignette("ggplot2-in-packages")` for more information.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">UpSetR</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue to the authors.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once per session.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use `linewidth` instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">UpSetR</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue to the authors.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once per session.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; Warning: The `size` argument of `element_line()` is deprecated as of ggplot2 3.4.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use the `linewidth` argument instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">UpSetR</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue to the authors.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once per session.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span></code></pre></div>
<p><img src="getting-started_files/figure-html/upset-1.png" class="r-plt" alt="" width="1050"></p>
</div>
<div class="section level3">
<h3 id="assembly-dendrogram">Assembly Dendrogram<a class="anchor" aria-label="anchor" href="#assembly-dendrogram"></a>
</h3>
<p>Cluster assemblies by their shared orthogroup content:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_dendro.html">plot_dendro</a></span><span class="op">(</span><span class="va">pa</span>, distance_method <span class="op">=</span> <span class="st">"jaccard"</span><span class="op">)</span></span></code></pre></div>
<p><img src="getting-started_files/figure-html/dendro-1.png" class="r-plt" alt="" width="1050"></p>
</div>
</div>
<div class="section level2">
<h2 id="working-with-scores">Working with Scores<a class="anchor" aria-label="anchor" href="#working-with-scores"></a>
</h2>
<p>If your data includes effector prediction scores (from omnieff), you
can:</p>
<div class="section level3">
<h3 id="filter-by-score">Filter by Score<a class="anchor" aria-label="anchor" href="#filter-by-score"></a>
</h3>
<p>Only include high-confidence effector predictions:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Build matrix with score threshold</span></span>
<span><span class="va">pa_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_pa_matrix.html">build_pa_matrix</a></span><span class="op">(</span></span>
<span>  <span class="va">clusters</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"binary"</span>,</span>
<span>  score_threshold <span class="op">=</span> <span class="fl">5.0</span>,  <span class="co"># Only include proteins with score &gt;= 5</span></span>
<span>  proteins <span class="op">=</span> <span class="va">proteins</span>     <span class="co"># Need proteins to access scores</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="visualize-score-distributions">Visualize Score Distributions<a class="anchor" aria-label="anchor" href="#visualize-score-distributions"></a>
</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Score distributions across all assemblies</span></span>
<span><span class="fu"><a href="../reference/plot_scores.html">plot_scores</a></span><span class="op">(</span><span class="va">proteins</span><span class="op">)</span></span></code></pre></div>
<p><img src="getting-started_files/figure-html/scores-1.png" class="r-plt" alt="" width="1050"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Faceted by assembly with threshold line</span></span>
<span><span class="fu"><a href="../reference/plot_scores.html">plot_scores</a></span><span class="op">(</span><span class="va">proteins</span>, by_assembly <span class="op">=</span> <span class="cn">TRUE</span>, threshold <span class="op">=</span> <span class="fl">5.0</span><span class="op">)</span></span></code></pre></div>
<p><img src="getting-started_files/figure-html/scores-2.png" class="r-plt" alt="" width="1050"></p>
</div>
</div>
<div class="section level2">
<h2 id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<p>The core paneffectR workflow:</p>
<ol style="list-style-type: decimal">
<li>
<code><a href="../reference/load_proteins.html">load_proteins()</a></code> - Load FASTA files (and optional
scores)</li>
<li>
<code><a href="../reference/cluster_proteins.html">cluster_proteins()</a></code> - Group proteins into
orthogroups</li>
<li>
<code><a href="../reference/build_pa_matrix.html">build_pa_matrix()</a></code> - Create presence/absence matrix</li>
<li>
<code>plot_*()</code> - Visualize results</li>
</ol>
</div>
<div class="section level2">
<h2 id="next-steps">Next Steps<a class="anchor" aria-label="anchor" href="#next-steps"></a>
</h2>
<ul>
<li>
<strong><a href="effector-analysis.html">Effector
Analysis</a></strong> - Working with omnieff output and score
filtering</li>
<li>
<strong><a href="pan-genome.html">Pan-Genome Analysis</a></strong> -
Analyzing core vs accessory proteins</li>
<li>
<strong><a href="algorithms.html">Algorithm Deep Dive</a></strong> -
Technical details on clustering methods</li>
</ul>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://github.com/danmaclean" class="external-link">Dan MacLean</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
