<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Pan-Genome Analysis with paneffectR • paneffectR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Pan-Genome Analysis with paneffectR">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">paneffectR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes">
<li><a class="dropdown-item" href="../articles/getting-started.html">Getting Started</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Workflows</h6></li>
    <li><a class="dropdown-item" href="../articles/effector-analysis.html">Effector Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/pan-genome.html">Pan-Genome Analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Technical</h6></li>
    <li><a class="dropdown-item" href="../articles/algorithms.html">Algorithm Deep Dive</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/TeamMacLean/paneffectR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Pan-Genome Analysis with paneffectR</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/TeamMacLean/paneffectR/blob/main/vignettes/pan-genome.Rmd" class="external-link"><code>vignettes/pan-genome.Rmd</code></a></small>
      <div class="d-none name"><code>pan-genome.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>While paneffectR was designed for effector analysis, it works equally
well for <strong>general pan-genome comparisons</strong> with any
protein FASTA files. This vignette demonstrates how to:</p>
<ul>
<li>Compare protein content across genome assemblies</li>
<li>Identify core, accessory, and unique proteins</li>
<li>Visualize pan-genome structure</li>
</ul>
<p>No effector scores are required.</p>
</div>
<div class="section level2">
<h2 id="pan-genome-concepts">Pan-Genome Concepts<a class="anchor" aria-label="anchor" href="#pan-genome-concepts"></a>
</h2>
<p>A pan-genome represents the complete gene content across a set of
related genomes:</p>
<ul>
<li>
<strong>Core genome</strong>: Genes present in all (or nearly all)
genomes - essential functions</li>
<li>
<strong>Accessory genome</strong>: Genes present in some genomes -
niche adaptation, lifestyle</li>
<li>
<strong>Unique genes</strong>: Present in only one genome -
strain-specific features</li>
</ul>
<p>paneffectR helps you explore this structure through presence/absence
analysis.</p>
</div>
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/TeamMacLean/paneffectR" class="external-link">paneffectR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="loading-fasta-files">Loading FASTA Files<a class="anchor" aria-label="anchor" href="#loading-fasta-files"></a>
</h2>
<p>For pan-genome analysis, you only need FASTA files:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get path to test data</span></span>
<span><span class="va">testdata_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"testdata"</span>, package <span class="op">=</span> <span class="st">"paneffectR"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load FASTAs only (no scores)</span></span>
<span><span class="va">proteins</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_proteins.html">load_proteins</a></span><span class="op">(</span></span>
<span>  fasta_dir <span class="op">=</span> <span class="va">testdata_dir</span>,</span>
<span>  pattern <span class="op">=</span> <span class="st">"*.faa"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">proteins</span></span>
<span><span class="co">#&gt; -- protein_collection --</span></span>
<span><span class="co">#&gt; 3 assemblies, 300 total proteins</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 3</span></span></span>
<span><span class="co">#&gt;   assembly_name n_proteins has_scores</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> assembly1            100 FALSE     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> assembly2            100 FALSE     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> assembly3            100 FALSE</span></span></code></pre></div>
<p>Note: Even though our test data has scores available, we’re ignoring
them for this analysis.</p>
<div class="section level3">
<h3 id="protein-id-requirements">Protein ID Requirements<a class="anchor" aria-label="anchor" href="#protein-id-requirements"></a>
</h3>
<p>Protein IDs must be <strong>unique across all assemblies</strong>. If
your FASTAs have generic headers like <code>&gt;protein_001</code>,
you’ll need to prefix them with assembly names.</p>
<p>paneffectR validates this automatically:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># This would fail if IDs weren't unique</span></span>
<span><span class="co"># (our test data already has unique IDs like "assembly1_000001")</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">proteins</span><span class="op">$</span><span class="va">assemblies</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">proteins</span><span class="op">$</span><span class="va">protein_id</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "assembly1_000039" "assembly1_000040" "assembly1_000094" "assembly1_000122"</span></span>
<span><span class="co">#&gt; [5] "assembly1_000164" "assembly1_000194"</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="clustering-proteins">Clustering Proteins<a class="anchor" aria-label="anchor" href="#clustering-proteins"></a>
</h2>
<p>Group proteins from different assemblies into orthogroups:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Cluster using DIAMOND reciprocal best hits</span></span>
<span><span class="va">clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cluster_proteins.html">cluster_proteins</a></span><span class="op">(</span><span class="va">proteins</span>, method <span class="op">=</span> <span class="st">"diamond_rbh"</span><span class="op">)</span></span></code></pre></div>
<p>For this vignette, we’ll use pre-computed clusters:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">visual_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"testdata"</span>, <span class="st">"visual"</span>, package <span class="op">=</span> <span class="st">"paneffectR"</span><span class="op">)</span></span>
<span><span class="va">clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">visual_dir</span>, <span class="st">"clusters_visual.rds"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">clusters</span></span>
<span><span class="co">#&gt; -- orthogroup_result (synthetic_visual) --</span></span>
<span><span class="co">#&gt; 50 orthogroups</span></span></code></pre></div>
<div class="section level3">
<h3 id="understanding-the-results">Understanding the Results<a class="anchor" aria-label="anchor" href="#understanding-the-results"></a>
</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Orthogroup statistics</span></span>
<span><span class="va">clusters</span><span class="op">$</span><span class="va">stats</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 × 4</span></span></span>
<span><span class="co">#&gt;   n_orthogroups n_singletons n_proteins_clustered n_assemblies</span></span>
<span><span class="co">#&gt;           <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span>            50            0                  160            6</span></span>
<span></span>
<span><span class="co"># How many singletons (unique proteins)?</span></span>
<span><span class="fu"><a href="../reference/n_singletons.html">n_singletons</a></span><span class="op">(</span><span class="va">clusters</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0</span></span>
<span></span>
<span><span class="co"># Singletons per assembly</span></span>
<span><span class="fu"><a href="../reference/singletons_by_assembly.html">singletons_by_assembly</a></span><span class="op">(</span><span class="va">clusters</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 0 × 2</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 2 variables: assembly &lt;chr&gt;, n_singletons &lt;int&gt;</span></span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="building-the-presenceabsence-matrix">Building the Presence/Absence Matrix<a class="anchor" aria-label="anchor" href="#building-the-presenceabsence-matrix"></a>
</h2>
<p>Create a binary matrix showing which orthogroups are in which
assemblies:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pa</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_pa_matrix.html">build_pa_matrix</a></span><span class="op">(</span><span class="va">clusters</span>, type <span class="op">=</span> <span class="st">"binary"</span><span class="op">)</span></span>
<span><span class="va">pa</span></span>
<span><span class="co">#&gt; -- pa_matrix (binary) --</span></span>
<span><span class="co">#&gt; 50 orthogroups x 6 assemblies</span></span>
<span><span class="co">#&gt; Sparsity: 46.7%</span></span>
<span></span>
<span><span class="co"># Include singletons (default) or exclude them</span></span>
<span><span class="va">pa_no_singletons</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_pa_matrix.html">build_pa_matrix</a></span><span class="op">(</span></span>
<span>  <span class="va">clusters</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"binary"</span>,</span>
<span>  exclude_singletons <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"With singletons:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">pa</span><span class="op">$</span><span class="va">matrix</span><span class="op">)</span>, <span class="st">"orthogroups\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; With singletons: 50 orthogroups</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Without singletons:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">pa_no_singletons</span><span class="op">$</span><span class="va">matrix</span><span class="op">)</span>, <span class="st">"orthogroups\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Without singletons: 50 orthogroups</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="analyzing-pan-genome-structure">Analyzing Pan-Genome Structure<a class="anchor" aria-label="anchor" href="#analyzing-pan-genome-structure"></a>
</h2>
<div class="section level3">
<h3 id="core-genome">Core Genome<a class="anchor" aria-label="anchor" href="#core-genome"></a>
</h3>
<p>Proteins present in all assemblies:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n_assemblies</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">pa</span><span class="op">$</span><span class="va">matrix</span><span class="op">)</span></span>
<span><span class="va">presence_count</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">pa</span><span class="op">$</span><span class="va">matrix</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Core = present in all</span></span>
<span><span class="va">core_og</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">presence_count</span><span class="op">[</span><span class="va">presence_count</span> <span class="op">==</span> <span class="va">n_assemblies</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Core orthogroups:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">core_og</span><span class="op">)</span>, <span class="st">"of"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">presence_count</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Core orthogroups: 10 of 50</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Core percentage:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fl">100</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">core_og</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">presence_count</span><span class="op">)</span>, <span class="fl">1</span><span class="op">)</span>, <span class="st">"%\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Core percentage: 20 %</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="accessory-genome">Accessory Genome<a class="anchor" aria-label="anchor" href="#accessory-genome"></a>
</h3>
<p>Proteins present in some but not all assemblies:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Accessory = present in 2 to (n-1) assemblies</span></span>
<span><span class="va">accessory_og</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">presence_count</span><span class="op">[</span><span class="va">presence_count</span> <span class="op">&gt;</span> <span class="fl">1</span> <span class="op">&amp;</span> <span class="va">presence_count</span> <span class="op">&lt;</span> <span class="va">n_assemblies</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Accessory orthogroups:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">accessory_og</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Accessory orthogroups: 30</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Accessory percentage:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fl">100</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">accessory_og</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">presence_count</span><span class="op">)</span>, <span class="fl">1</span><span class="op">)</span>, <span class="st">"%\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Accessory percentage: 60 %</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="unique-genes">Unique Genes<a class="anchor" aria-label="anchor" href="#unique-genes"></a>
</h3>
<p>Proteins present in only one assembly (including singletons):</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">unique_og</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">presence_count</span><span class="op">[</span><span class="va">presence_count</span> <span class="op">==</span> <span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Unique orthogroups:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">unique_og</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Unique orthogroups: 10</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Unique percentage:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fl">100</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">unique_og</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">presence_count</span><span class="op">)</span>, <span class="fl">1</span><span class="op">)</span>, <span class="st">"%\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Unique percentage: 20 %</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="pan-genome-summary">Pan-Genome Summary<a class="anchor" aria-label="anchor" href="#pan-genome-summary"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pan_summary</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  Category <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Core"</span>, <span class="st">"Accessory"</span>, <span class="st">"Unique"</span>, <span class="st">"Total"</span><span class="op">)</span>,</span>
<span>  Count <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">core_og</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">accessory_og</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">unique_og</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">presence_count</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  Percentage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fl">100</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">core_og</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">presence_count</span><span class="op">)</span>, <span class="fl">1</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fl">100</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">accessory_og</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">presence_count</span><span class="op">)</span>, <span class="fl">1</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fl">100</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">unique_og</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">presence_count</span><span class="op">)</span>, <span class="fl">1</span><span class="op">)</span>,</span>
<span>    <span class="fl">100</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">pan_summary</span></span>
<span><span class="co">#&gt;    Category Count Percentage</span></span>
<span><span class="co">#&gt; 1      Core    10         20</span></span>
<span><span class="co">#&gt; 2 Accessory    30         60</span></span>
<span><span class="co">#&gt; 3    Unique    10         20</span></span>
<span><span class="co">#&gt; 4     Total    50        100</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="visualizing-the-pan-genome">Visualizing the Pan-Genome<a class="anchor" aria-label="anchor" href="#visualizing-the-pan-genome"></a>
</h2>
<div class="section level3">
<h3 id="heatmap">Heatmap<a class="anchor" aria-label="anchor" href="#heatmap"></a>
</h3>
<p>The heatmap shows presence (colored) and absence (white)
patterns:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ht</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span></span>
<span>  <span class="va">pa</span>,</span>
<span>  cluster_rows <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  cluster_cols <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  show_row_names <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"white"</span>, <span class="st">"steelblue"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu">ComplexHeatmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/draw-dispatch.html" class="external-link">draw</a></span><span class="op">(</span><span class="va">ht</span><span class="op">)</span></span></code></pre></div>
<p><img src="pan-genome_files/figure-html/heatmap-1.png" class="r-plt" alt="" width="1050"></p>
</div>
<div class="section level3">
<h3 id="upset-plot">UpSet Plot<a class="anchor" aria-label="anchor" href="#upset-plot"></a>
</h3>
<p>UpSet plots excel at showing intersections - which orthogroups are
shared between which combinations of assemblies:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Show all intersections with at least 1 orthogroup</span></span>
<span><span class="fu"><a href="../reference/plot_upset.html">plot_upset</a></span><span class="op">(</span><span class="va">pa</span>, min_size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: `aes_string()` was deprecated in ggplot2 3.0.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use tidy evaluation idioms with `aes()`.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> See also `vignette("ggplot2-in-packages")` for more information.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">UpSetR</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue to the authors.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once per session.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use `linewidth` instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">UpSetR</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue to the authors.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once per session.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; Warning: The `size` argument of `element_line()` is deprecated as of ggplot2 3.4.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use the `linewidth` argument instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">UpSetR</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue to the authors.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once per session.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span></code></pre></div>
<p><img src="pan-genome_files/figure-html/upset-1.png" class="r-plt" alt="" width="1050"></p>
<p>The bars show: - How many orthogroups are unique to each assembly -
How many are shared between specific combinations - How many are in the
core (all assemblies)</p>
</div>
<div class="section level3">
<h3 id="assembly-dendrogram">Assembly Dendrogram<a class="anchor" aria-label="anchor" href="#assembly-dendrogram"></a>
</h3>
<p>Cluster assemblies by their shared protein content:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_dendro.html">plot_dendro</a></span><span class="op">(</span><span class="va">pa</span>, distance_method <span class="op">=</span> <span class="st">"jaccard"</span><span class="op">)</span></span></code></pre></div>
<p><img src="pan-genome_files/figure-html/dendro-1.png" class="r-plt" alt="" width="1050"></p>
<p>Try different distance metrics:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Jaccard: good for presence/absence</span></span>
<span><span class="fu"><a href="../reference/plot_dendro.html">plot_dendro</a></span><span class="op">(</span><span class="va">pa</span>, distance_method <span class="op">=</span> <span class="st">"jaccard"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Jaccard Distance"</span><span class="op">)</span></span></code></pre></div>
<p><img src="pan-genome_files/figure-html/dendro-methods-1.png" class="r-plt" alt="" width="1050"></p>
</div>
</div>
<div class="section level2">
<h2 id="comparing-assemblies">Comparing Assemblies<a class="anchor" aria-label="anchor" href="#comparing-assemblies"></a>
</h2>
<div class="section level3">
<h3 id="pairwise-similarity">Pairwise Similarity<a class="anchor" aria-label="anchor" href="#pairwise-similarity"></a>
</h3>
<p>Calculate Jaccard similarity between all pairs:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Simple Jaccard calculation</span></span>
<span><span class="va">jaccard</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">a</span>, <span class="va">b</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">intersection</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">a</span> <span class="op">==</span> <span class="fl">1</span> <span class="op">&amp;</span> <span class="va">b</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span></span>
<span>  <span class="va">union</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">a</span> <span class="op">==</span> <span class="fl">1</span> <span class="op">|</span> <span class="va">b</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span></span>
<span>  <span class="va">intersection</span> <span class="op">/</span> <span class="va">union</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Calculate for all pairs</span></span>
<span><span class="va">assemblies</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pa</span><span class="op">$</span><span class="va">matrix</span><span class="op">)</span></span>
<span><span class="va">n</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">assemblies</span><span class="op">)</span></span>
<span><span class="va">sim_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fl">0</span>, <span class="va">n</span>, <span class="va">n</span>, dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">assemblies</span>, <span class="va">assemblies</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="va">n</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="kw">for</span> <span class="op">(</span><span class="va">j</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="va">n</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">sim_matrix</span><span class="op">[</span><span class="va">i</span>, <span class="va">j</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">jaccard</span><span class="op">(</span><span class="va">pa</span><span class="op">$</span><span class="va">matrix</span><span class="op">[</span>, <span class="va">i</span><span class="op">]</span>, <span class="va">pa</span><span class="op">$</span><span class="va">matrix</span><span class="op">[</span>, <span class="va">j</span><span class="op">]</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">sim_matrix</span>, <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">#&gt;       asm_A asm_B asm_C asm_D asm_E asm_F</span></span>
<span><span class="co">#&gt; asm_A  1.00  0.49  0.50  0.53  0.44  0.33</span></span>
<span><span class="co">#&gt; asm_B  0.49  1.00  0.54  0.57  0.53  0.38</span></span>
<span><span class="co">#&gt; asm_C  0.50  0.54  1.00  0.46  0.46  0.43</span></span>
<span><span class="co">#&gt; asm_D  0.53  0.57  0.46  1.00  0.49  0.42</span></span>
<span><span class="co">#&gt; asm_E  0.44  0.53  0.46  0.49  1.00  0.59</span></span>
<span><span class="co">#&gt; asm_F  0.33  0.38  0.43  0.42  0.59  1.00</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="which-assemblies-share-the-most">Which Assemblies Share the Most?<a class="anchor" aria-label="anchor" href="#which-assemblies-share-the-most"></a>
</h3>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Find most similar pair (excluding diagonal)</span></span>
<span><span class="va">sim_matrix_upper</span> <span class="op">&lt;-</span> <span class="va">sim_matrix</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/diag.html" class="external-link">diag</a></span><span class="op">(</span><span class="va">sim_matrix_upper</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fl">0</span></span>
<span><span class="va">max_idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">sim_matrix_upper</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">sim_matrix_upper</span><span class="op">)</span>, arr.ind <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span>, <span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Most similar pair:"</span>,</span>
<span>    <span class="va">assemblies</span><span class="op">[</span><span class="va">max_idx</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="st">"and"</span>, <span class="va">assemblies</span><span class="op">[</span><span class="va">max_idx</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span>,</span>
<span>    <span class="st">"\nJaccard similarity:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">sim_matrix_upper</span><span class="op">)</span>, <span class="fl">3</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Most similar pair: asm_F and asm_E </span></span>
<span><span class="co">#&gt; Jaccard similarity: 0.588</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="assembly-specific-analysis">Assembly-Specific Analysis<a class="anchor" aria-label="anchor" href="#assembly-specific-analysis"></a>
</h2>
<div class="section level3">
<h3 id="proteins-unique-to-one-assembly">Proteins Unique to One Assembly<a class="anchor" aria-label="anchor" href="#proteins-unique-to-one-assembly"></a>
</h3>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get singletons with assembly information</span></span>
<span><span class="va">singletons_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/singletons_by_assembly.html">singletons_by_assembly</a></span><span class="op">(</span><span class="va">clusters</span><span class="op">)</span></span>
<span><span class="va">singletons_df</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 0 × 2</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 2 variables: assembly &lt;chr&gt;, n_singletons &lt;int&gt;</span></span></span>
<span></span>
<span><span class="co"># Which assembly has the most unique proteins?</span></span>
<span><span class="va">singletons_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">n_singletons</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 0 × 2</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 2 variables: assembly &lt;chr&gt;, n_singletons &lt;int&gt;</span></span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="orthogroups-missing-from-one-assembly">Orthogroups Missing from One Assembly<a class="anchor" aria-label="anchor" href="#orthogroups-missing-from-one-assembly"></a>
</h3>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Which orthogroups are missing from asm_A?</span></span>
<span><span class="va">missing_from_asm_A</span> <span class="op">&lt;-</span> <span class="va">pa</span><span class="op">$</span><span class="va">matrix</span><span class="op">[</span><span class="va">pa</span><span class="op">$</span><span class="va">matrix</span><span class="op">[</span>, <span class="st">"asm_A"</span><span class="op">]</span> <span class="op">==</span> <span class="fl">0</span>, , drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Orthogroups missing from asm_A:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">missing_from_asm_A</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Orthogroups missing from asm_A: 27</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="exporting-results">Exporting Results<a class="anchor" aria-label="anchor" href="#exporting-results"></a>
</h2>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Export presence/absence matrix as CSV</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">pa</span><span class="op">)</span>,</span>
<span>  <span class="st">"pan_genome_matrix.csv"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Export orthogroup membership</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span></span>
<span>  <span class="va">clusters</span><span class="op">$</span><span class="va">orthogroups</span>,</span>
<span>  <span class="st">"orthogroup_membership.csv"</span>,</span>
<span>  row.names <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Export pan-genome summary</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span></span>
<span>  <span class="va">pan_summary</span>,</span>
<span>  <span class="st">"pan_genome_summary.csv"</span>,</span>
<span>  row.names <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<p>Key steps for pan-genome analysis:</p>
<ol style="list-style-type: decimal">
<li>Load FASTA files with <code>load_proteins(fasta_dir)</code>
</li>
<li>Cluster into orthogroups with <code><a href="../reference/cluster_proteins.html">cluster_proteins()</a></code>
</li>
<li>Build binary matrix with
<code>build_pa_matrix(type = "binary")</code>
</li>
<li>Analyze core/accessory/unique proportions</li>
<li>Visualize with <code><a href="../reference/plot_heatmap.html">plot_heatmap()</a></code>,
<code><a href="../reference/plot_upset.html">plot_upset()</a></code>, <code><a href="../reference/plot_dendro.html">plot_dendro()</a></code>
</li>
</ol>
</div>
<div class="section level2">
<h2 id="next-steps">Next Steps<a class="anchor" aria-label="anchor" href="#next-steps"></a>
</h2>
<ul>
<li>
<strong><a href="getting-started.html">Getting Started</a></strong>
- Basic workflow overview</li>
<li>
<strong><a href="effector-analysis.html">Effector
Analysis</a></strong> - Working with omnieff scores</li>
<li>
<strong><a href="algorithms.html">Algorithm Deep Dive</a></strong> -
Technical details on clustering</li>
</ul>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://github.com/danmaclean" class="external-link">Dan MacLean</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
